Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -621,50 +612,34 @@ private void RegisterCommandMenuItems(IWindsorContainer container)
                 .If(type => type.NotDisabledOrExperimental(_initialSettings))
-                .WithService.Base() 
-                .Configure(item => item.DependsOn(Dependency.OnComponent(typeof(CommandBase),
-                    CommandNameFromCommandMenuName(item.Implementation.Name))))
+                .WithService.Base()
                 .LifestyleTransient());
         }
 
-        private string CommandNameFromCommandMenuName(string itemName)
-        {
-            //note: CommandBase naming convention: [Foo]Command
-            //note: ICommandMenuItem naming convention for [Foo]Command: [Foo]CommandMenuItem
-            return itemName.Substring(0, itemName.Length - "MenuItem".Length);
-        }
-
         private void RegisterCommands(IWindsorContainer container)
         {
             //note: convention: the registration name for commands is the type name, not the full type name.
             //Otherwise, namespaces would get in the way when binding to the menu items.
             RegisterCommandsWithPresenters(container);
 
-            // assumption: All Commands (and CommandMenuItems by extension) are in the same assembly as CommandBase
-            var commandsForCommandMenuItems = Assembly.GetAssembly(typeof(CommandBase)).GetTypes()
-                .Where(type => type.IsClass 
-                               && typeof(ICommandMenuItem).IsAssignableFrom(type) 
-                               && type.NotDisabledOrExperimental(_initialSettings))
-                .Select(type => CommandNameFromCommandMenuName(type.Name))
-                .ToHashSet();
-
-            container.Register(Classes.FromAssemblyContaining<CommandBase>()
-                .Where(type => type.Namespace != null
-                            && type.Namespace.StartsWith(typeof(CommandBase).Namespace ?? string.Empty)
-                            && (type.BaseType == typeof(CommandBase) || type.BaseType == typeof(RefactorCommandBase))
-                            && type.Name.EndsWith("Command")
-                            && type.NotDisabledOrExperimental(_initialSettings)
-                            && commandsForCommandMenuItems.Contains(type.Name))
-                .WithService.Self()
+            // assumption: All Commands are in the same assembly as CommandBase
+            container.Register(Classes.FromAssemblyContaining(typeof(CommandBase))
+                .IncludeNonPublicTypes()
+                .Where(type => type.IsBasedOn(typeof(CommandBase))
+                    && type != typeof(DelegateCommand) // DelegateCommand is not intended to be injected!
+                    && type.NotDisabledOrExperimental(_initialSettings))
                 .WithService.Select(new[] { typeof(CommandBase) })
-                .LifestyleTransient()
-                .Configure(c => c.Named(c.Implementation.Name)));
+                .WithService.Self()
+                .WithService.DefaultInterfaces()
+                .LifestyleTransient());
```

Comment:
I only got to see this now and I do not really like this registration. I think it should use a `BasedOn` followed by an `If` for the remaining conditions and a `WithService.Base`.

Moreover, do we really need the default interface registration? 

Furthermore, are we sure that we can drop the naming of the registrations? The default names are painful to use in naming conventions because they include the namespace.